examples/cpp-dummy/micro_cpp_dummy.cpp

```diff
@@ -0,0 +1,111 @@
+// Micro simulation
+// In this script we solve a dummy micro problem to show how to adjust the macro-micro coupling
+// This dummy is written in C++ and is bound to python using pybind11
+#include <iostream>
+#include <vector>
+#include <pybind11/pybind11.h>
+#include <pybind11/numpy.h> // numpy arrays
+#include <pybind11/stl.h> // std::vector conversion
+
+namespace py = pybind11;
+
+class MicroSimulation
+{
+public:
+    MicroSimulation(int sim_id);
+    void initialize();
+    // solve takes python dict for macro_write data, dt, and returns python dict for macro_read data
+    py::dict solve(py::dict macro_write_data, double dt);
+    void save_checkpoint();
+    void reload_checkpoint();
+    int get_dims();
+
+private:
+    int _sim_id;
+    int _dims;
+    double _micro_scalar_data;
+    std::vector<double> _micro_vector_data;
+    double _checkpoint;
+};
```

Comment:
Does it make more C++ sense to move this to a `micro_cpp_dummy.hpp` file? If yes, lets do it here too, to be consistent with what a developer in C++ would do.